src/main/java/io/quarkus/develocity/project/QuarkusProjectDevelocityConfigurationListener.java

```diff
@@ -42,23 +41,22 @@ public class QuarkusProjectDevelocityConfigurationListener implements GradleEnte
             File.separator + "target" + File.separator + "test-project" + File.separator);
 
     @Override
-    public void configure(GradleEnterpriseApi gradleEnterpriseApi, MavenSession mavenSession) throws Exception {
+    public void configure(DevelocityApi develocityApi, MavenSession mavenSession) throws Exception {
         if (ignoreProject(mavenSession)) {
             // do not publish a build scan for test builds
             Log.debug("Disabling build scan publication and build cache for nested project: "
                     + mavenSession.getRequest().getBaseDirectory());
 
-            gradleEnterpriseApi.getBuildScan().publishAlwaysIf(false);
-            gradleEnterpriseApi.getBuildScan().publishOnFailureIf(false);
-            gradleEnterpriseApi.getBuildCache().getLocal().setEnabled(false);
-            gradleEnterpriseApi.getBuildCache().getRemote().setEnabled(false);
+            develocityApi.getBuildScan().getPublishing().onlyIf(context -> !context.getBuildResult().getFailures().isEmpty());
```

Comment:
I'm not sure we want to do that. We really don't want anything published for these projects, even in the case of failure (the failure will be reported in the actual module).